content-src/components/StartupOverlay/StartupOverlay.jsx

```diff
@@ -15,8 +15,29 @@ export class _StartupOverlay extends React.PureComponent {
 
     this.state = {
       emailInput: "",
-      overlayRemoved: false
+      overlayRemoved: false,
+      flowId: "",
+      flowBeginTime: ""
     };
+    this.didFetch = false;
+  }
+
+  async componentWillUpdate() {
+    if (this.props.fxa_endpoint && !this.didFetch) {
+      try {
+        this.didFetch = true;
+        const response = await fetch(`${this.props.fxa_endpoint}/metrics-flow`);
```

Comment:
Looks like this is causing test failure timeout:
TEST-UNEXPECTED-FAIL | browser/base/content/test/urlbar/browser_urlbar_blanking.js | Found a tab after previous test timed out: about:welcome -
https://treeherder.mozilla.org/logviewer.html#?job_id=187870714&repo=try&lineNumber=1633

Where the test loads gInitialPages:
https://searchfox.org/mozilla-central/source/browser/base/content/browser.js#321

The test times out with about:welcome open and status bar showing "Performing TLS handshake to 127.0.0.1"

Reply:
Ah, during testing, it's set to {server} so 127.0.0.1, and the request silently fails:

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://127.0.0.1:8888//metrics-flow. (Reason: CORS request did not succeed).

Reply:
The following works because it looks like the test/metrics-flow endpoint just never resolves in time for the test:
```js
const controller = new AbortController();
setTimeout(() => controller.abort(), 5000);
const response = await fetch(`${this.props.fxa_endpoint}/metrics-flow`, {signal: controller.signal});
```

Unclear how long we should wait before aborting (and triggering a FXA_METRICS_ERROR…

Reply:
@mardak would you prefer something like that, or to edit the test?

Reply:
It's not just a single test, so you won't be able to edit the test

Reply:
true, it would fail anywhere this page is used wouldn't it